// fake-todo-list-js/index.js
var counter = 1;

var todoList=[
    {
        id:counter++,
        idCreator:2,
        idAssigne:1,
        action:"to study"
    },
    {
        id:counter++,
        idCreator:1,
        idAssigne:3,
        action:"read a book"
    },
    {
        id:counter++,
        idCreator:3,
        idAssigne:2,
        action:"play"
    },
   
]

var users=[
    {
        id:1,
        name:"Gino"
    },
    {
        id:2,
        name:"Franco"
    },
    {
        id:3,
        name:"Lina"
    }
]



exports.getTodo=function(){
    return todoList;
}


exports.addTodo=function(list) {
    list.id = counter++;
    todoList.push(list);
    return "insert action";
}


exports.deleteTodo=function(id){
    for(var i=0; i<todoList.length; i++){
        if (todoList[i].id===id) {
            todoList.splice(i, 1);
        }
      }
    return todoList;
}


exports.getComplete=function(id){
    for (var i = 0; i < todoList.length; i++) {
        if (todoList[i].id===id) {
            todoList[i].status=true
            return "Update True"
        }
    }
}
    

exports.getNotCompleted=function(id){
    for (var i = 0; i < todoList.length; i++) {
        if (todoList[i].id===id) {
            todoList[i].status=false
            return "Update False"
        }     
    }
}


exports.getUsers=function(){
    return users;
}


exports.addUsers=function(list) {
    list.id = counter++;
    users.push(list);
    return users;
}


exports.todoUser=function(id){
    var usersList=[]
    for (var i = 0; i < todoList.length; i++) {
        if ((todoList[i].idCreator===id) || (todoList[i].idAssigne===id)) {
            usersList.push(todoList[i])
        }
    }
    return usersList
}


exports.getActionTrue=function(){
    var actionComp=[]
    for (var i = 0; i <todoList.length; i++) {
        if (todoList[i].status===true) {
            actionComp.push({
                id:todoList[i].id,
                action:todoList[i].action,
                status:todoList[i].status
            })
        }
    }
    return actionComp;
}


exports.getActionFalse=function(){
    var actionNotComp=[]
    for (var i = 0; i <todoList.length; i++) {
        if (todoList[i].status===false) {
            actionNotComp.push({
                id:todoList[i].id,
                action:todoList[i].action,
                status:todoList[i].status
            })
        }
    }
    return actionNotComp;
}
